main/main.c

```diff
@@ -2,72 +2,112 @@
+//                          OAuthInterfaceHandler.Status, OAuthInterfaceHandler.token.AccessToken,
+//                          OAuthInterfaceHandler.OAuthBuffer.status, 
+//                          OAuthInterfaceHandler.OAuthBuffer.MessageBuffer);
+//     if (CommandResult == false)
+//     {
+//         ESP_LOGE(TAG, "Playback info update failed");
+//         return;
+//     }
+//     GUI_UpdateSpotifyScreen(SpotifyInterfaceHandler.PlaybackInfo->ArtistName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->SongName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->AlbumName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->Duration,
+//                             SpotifyInterfaceHandler.PlaybackInfo->Progress);
+//     ESP_LOGI(TAG, "Playback info updated");
+// }
+
+void app_main(void)//                          OAuthInterfaceHandler.OAuthBuffer.status, 
+//                          OAuthInterfaceHandler.OAuthBuffer.MessageBuffer);
+//     if (CommandResult == false)
+//     {
+//         ESP_LOGE(TAG, "Playback info update failed");
+//         return;
+//     }
+//     GUI_UpdateSpotifyScreen(SpotifyInterfaceHandler.PlaybackInfo->ArtistName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->SongName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->AlbumName,
+//                             SpotifyInterfaceHandler.PlaybackInfo->Duration,
+//                             SpotifyInterfaceHandler.PlaybackInfo->Progress);
+//     ESP_LOGI(TAG, "Playback info updated");
+// }
 {
-    GUI_TaskInit();
+    //GUI_TaskInit();
     GlobalInit();
     nvsFlashInit();
     SpiffsGlobalConfig();
 #ifdef WIFI_INIT_STA_MODE
-    WifiStationMode("Hardware10", "87654321");
-    // WifiStationMode("BELL789", "167271A164A9");
+    // WifiStationMode("Done_IOT_1", "87654321@");
+    WifiStationMode("BELL789", "167271A164A9");
 #else
     wifiConnectionModule();
 #endif
 
 #ifdef SpotifyEnable
 
-    SpotifyInterfaceHandler.IsSpotifyAuthorizedSemaphore = &IsSpotifyAuthorizedSemaphore;
-    SpotifyInterfaceHandler.ConfigAddressInSpiffs = SpotifyConfigAddressInSpiffs;
-    Spotify_TaskInit(&SpotifyInterfaceHandler);
+//TODO merge all in a static function
+    OAuthInterfaceHandler.IsServiceAuthorizedSemaphore = &IsSpotifyAuthorizedSemaphore;
+    OAuthInterfaceHandler.ConfigAddressInSpiffs = SpotifyConfigAddressInSpiffs;
+    Spotify_ClientInfo = Spotify_ClientConfigInit();
+    OAuthInterfaceHandler.ClientConfig.url = Spotify_ClientInfo.url;
+    OAuthInterfaceHandler.ClientConfig.host = Spotify_ClientInfo.host;
+    OAuthInterfaceHandler.ClientConfig.path = Spotify_ClientInfo.path;
+    OAuthInterfaceHandler.ClientConfig.requestURI = Spotify_ClientInfo.requestURI;
+    OAuthInterfaceHandler.ClientConfig.responseURI = Spotify_ClientInfo.responseURI;
+    OAuthInterfaceHandler.ClientConfig.hostname = Spotify_ClientInfo.hostname;
```

Comment:
what is different between "Spotify_ClientInfo.host" and "Spotify_ClientInfo.hostname" please choose other name,actually I'm confuse

Reply:
the hostname used for MDNS stuff...fixed.